IsoLib/libisomediafile/src/MovieFragmentAtom.c

```diff
@@ -34,7 +34,8 @@ static void destroy(MP4AtomPtr s)
 
   self = (MP4MovieFragmentAtomPtr)s;
   if(self == NULL) BAILWITHERROR(MP4BadParamErr)
-  DESTROY_ATOM_LIST_F(atomList);
+  err = MP4DeleteLinkedList(self->atomList);
```

Comment:
This is already a part of the macro above.

Reply:
In side of Macro code 
```
#define DESTROY_ATOM_LIST                                      \
  if(self->atomList)                                           \
  {                                                            \
    u32 atomListSize;                                          \
    err = MP4GetListEntryCount(self->atomList, &atomListSize); \
    if(err) goto bail;                                         \
    for(i = 0; i < atomListSize; i++)                          \
    {                                                          \
      MP4AtomPtr a;                                            \
      err = MP4GetListEntry(self->atomList, i, (char **)&a);   \
      if(err) goto bail;                                       \
      if(a) a->destroy(a);                                     \
    }                                                          \
    err = MP4DeleteLinkedList(self->atomList);                 \
    if(err) goto bail;                                         \
  }
```

```if(a) a->destroy(a); ```  
```err = MP4DeleteLinkedList(self->atomList); ```   
I think that those free the memory same thing  
Is it Right or not ?  

My Thought is we use only ```err = MP4DeleteLinkedList(self->atomList); ``` this part for free the self->atomList   memory.  

I do not understand LinkedList have several ListEntry and several entry count.    

I add this code due to 'Segmentation Fault Error Raised'   
when I free the MP4MovieFragmentAtomPtr